src/main/java/ruproject/services/CarreraServiceImpl.java

```diff
@@ -42,12 +54,45 @@ public CarreraDTO saveCarrera(CarreraDTO carreraDTO) {
 
     @Override
     public CarreraDTO updateCarrera(String name, CarreraDTO carreraDTO) {
-        if((existsByName(name))){
-        Carrera carrera = carreraMapper.carreraDTOtoCarrera(carreraDTO);
-        carrera.setName(name);
-        return carreraMapper.carreraToCarreaDTO(carreraRepository.save(carrera));
+        if ((existsByName(name))) {
+            Carrera savedCarrera = carreraRepository.findByName(name);
+            Carrera returnCarrera = carreraMapper.carreraDTOtoCarrera(carreraDTO);
+            returnCarrera.setName(savedCarrera.getName());
+            returnCarrera.setId(savedCarrera.getId());
+
+            List<Materia> returnCarreraMaterias = returnCarrera.getMaterias();
+            List<Materia> savedCarrearMaterias = savedCarrera.getMaterias();
+            List<Materia> materias= new ArrayList<>();
+
+            if(savedCarrearMaterias.size()>0){
```

Comment:
Puedes usar el método isEmpty: https://www.geeksforgeeks.org/list-isempty-method-in-java-with-examples/